relayer/chains/cosmos/message_handlers_test.go

```diff
@@ -0,0 +1,299 @@
+package cosmos
+
+import (
+	"os"
+	"testing"
+
+	clienttypes "github.com/cosmos/ibc-go/v3/modules/core/02-client/types"
+	chantypes "github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
+	"github.com/cosmos/relayer/v2/relayer/processor"
+	"github.com/cosmos/relayer/v2/relayer/provider/cosmos"
+	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+)
+
+func mockCosmosChainProcessor(t *testing.T) *CosmosChainProcessor {
+	const (
+		chainID1 = "test-chain-1"
+		chainID2 = "test-chain-2"
+	)
+	var (
+		pathEnd1      = processor.PathEnd{ChainID: chainID1}
+		pathEnd2      = processor.PathEnd{ChainID: chainID2}
+		log           = zap.NewNop()
+		pathProcessor = processor.NewPathProcessor(log, pathEnd1, pathEnd2)
+		provider      = cosmos.CosmosProvider{PCfg: cosmos.CosmosProviderConfig{ChainID: chainID1}}
+	)
+
+	ccp, err := NewCosmosChainProcessor(log, &provider, "", os.Stdin, os.Stdout, []*processor.PathProcessor{pathProcessor})
+	require.NoError(t, err, "error constructing cosmos chain processor")
+
+	applicable := pathProcessor.SetChainProviderIfApplicable(&provider)
+	require.True(t, applicable, "error setting path processor reference to chain processor")
+
+	return ccp
```

Comment:
It's a minimally constructed `CosmosChainProcessor` used for testing purposes. The provider with the `ChainID` is necessary, for example, so that the `IsRelevantChannel` checks work.